middle_end/flambda2/simplify/simplify_variadic_primitive.ml

```diff
@@ -56,19 +107,28 @@ let simplify_make_block ~original_prim ~field_kind tag ~shape
               TE.add_env_extension typing_env env_extension))
     in
     let ty =
-      let fields =
-        List.map2
-          (fun arg kind_with_subkind ->
-            T.alias_type_of (K.With_subkind.kind kind_with_subkind) arg)
-          args shape
-      in
-      let alloc_mode = Alloc_mode.For_allocations.as_type alloc_mode in
-      match mutable_or_immutable with
-      | Immutable ->
-        T.immutable_block ~is_unique:false tag ~field_kind alloc_mode ~fields
-      | Immutable_unique ->
-        T.immutable_block ~is_unique:true tag ~field_kind alloc_mode ~fields
-      | Mutable -> T.mutable_block alloc_mode
+      match block_shape with
+      | Mixed _ ->
+        (* CR mixed blocks: I don't really know what I'm doing here. In
+           particular for [ty] I've picked [T.any_block] because exisiting more
+           specific types for blocks all want the fields to have the same kind.
+           I don't even know of [any_block] is correct -- perhaps it's meant to
+           be normal blocks below no scan tag. *)
+        T.any_block
```

Comment:
Going by the following code, looks like you can construct the same fields list and use the match on mutable_or_immutable for both cases?

Reply:
Yes, `T.any_block` is meant for blocks with regular constructors. Until we have support for mixed blocks in the Flambda2 type system, you should use `T.any_value`.

Reply:
@TheNumbat: I'm not sure I understand your suggestion. I'm taking Vincent's suggestion to use `T.any_value` here; let me know if it seems like I'm missing something.

Reply:
By the way, do you have a dedicated primitive for projecting regular values out of mixed blocks or do you use the existing `Pfield` primitive ? If you have a dedicated primitive then we can postpone any changes to the Flambda2 type system, if you reuse the existing primitive we will likely have to do at least some quick fixes to avoid wrongly confusing a mixed block and a regular block.

Reply:
Per discussion, I'm going to change this PR to use a dedicated primitive for projecting a regular value out of a mixed block.

Reply:
I now use the mixed-block version of the primitive for getting/setting fields in the value prefix as well.